include/tscore/TSStatus.h

```diff
@@ -0,0 +1,90 @@
+/**
+  @file TSStatus.h
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+#pragma once
+
+#include <tscore/ink_assert.h>
+
+// Global status information for trafficserver
+//
+class TsStatus
+{
+private:
+  struct Data {
+    bool ssl_handshaking_stopped;
+    bool event_system_shut_down;
+    bool draining;
+  };
+
+public:
+  static bool
+  is_ssl_handshaking_stopped()
+  {
+    return _d().ssl_handshaking_stopped;
+  }
+
+  static bool
+  is_event_system_shut_down()
+  {
+    return _d().event_system_shut_down;
+  }
+
+  // Keeps track if the server is in draining state, follows the proxy.node.config.draining metric.
+  //
+  static bool
+  is_draining()
+  {
+    return _d().draining;
+  }
+
+  static void
+  stop_ssl_handshaking()
+  {
+    ink_assert(!_d().ssl_handshaking_stopped);
+
+    _d().ssl_handshaking_stopped = true;
+  }
+
+  static void
+  shut_down_event_system()
+  {
+    ink_assert(_d().ssl_handshaking_stopped && !_d().event_system_shut_down);
+
+    _d().event_system_shut_down = true;
+  }
+
+  static void
+  drain(bool enable)
+  {
+    _d().draining = enable;
+  }
+
+private:
+  static Data &
+  _d()
```

Comment:
Really don't like short non descriptive names.  Since this is a singleton, call it _instance().